drivers/i3c/i3c_npcx.c

```diff
@@ -0,0 +1,2047 @@
+			} else {
+				/* Update target descriptor */
+				target->dynamic_addr = dyn_addr;
+				target->bcr = rx_buf[6];
+				target->dcr = rx_buf[7];
+			}
+
+			/* Mark the address as I3C device */
+			i3c_addr_slots_mark_i3c(&data->common.attached_dev.addr_slots, dyn_addr);
+
+			/*
+			 * If the device has static address, after address assignment,
+			 * the device will not respond to the static address anymore.
+			 * So free the static one from address slots if different from
+			 * newly assigned one.
+			 */
+			if ((target != NULL) && (target->static_addr != 0U) &&
+			    (dyn_addr != target->static_addr)) {
+				i3c_addr_slots_mark_free(&data->common.attached_dev.addr_slots,
+							 dyn_addr);
+			}
+
+			/* Emit process DAA again to send the address to the device */
+			inst->MWDATAB = dyn_addr;
+			ret = npcx_i3c_request_daa(inst);
+			if (ret != 0) {
+				LOG_ERR("%s: Assign DA timeout", __func__);
+				break;
+			}
+
+			LOG_DBG("PID 0x%04x%08x assigned dynamic address 0x%02x", vendor_id,
+				part_no, dyn_addr);
+
+			/* Target did not accept the assigned DA, exit DAA */
+			if (IS_BIT_SET(inst->MSTATUS, NPCX_I3C_MSTATUS_NACKED)) {
+				ret = -EFAULT;
+				LOG_DBG("TGT NACK assigned DA %#x", dyn_addr);
+
+				/* Free the reserved DA */
+				i3c_addr_slots_mark_free(&data->common.attached_dev.addr_slots,
+							 dyn_addr);
+
+				/* 0 if address has not been assigned */
+				if (target != NULL) {
+					target->dynamic_addr = 0;
+				}
+
+				break;
+			}
+		}
+
+		/* Check all targets have been assigned DA and DAA complete */
+	} while ((!IS_BIT_SET(inst->MSTATUS, NPCX_I3C_MSTATUS_COMPLETE)) &&
+		 npcx_i3c_state_get(inst) != MSTATUS_STATE_IDLE);
+
+out_do_daa:
+	/* Exit DAA mode when error occurs */
+	if (ret != 0) {
+		npcx_i3c_request_emit_stop(inst);
+	}
+
+	/* Clear all flags. */
+	npcx_i3c_errwarn_clear_all(inst);
+	npcx_i3c_status_clear_all(inst);
+
+	/* Re-Enable I3C IRQ sources. */
+	npcx_i3c_interrupt_enable(inst, intmask);
+
+	npcx_i3c_fifo_flush(inst);
+	npcx_i3c_mutex_unlock(dev);
+
+	return ret;
+}
+
+/*
+ * brief:  Send Common Command Code (CCC).
+ *
+ * param[in] dev      Pointer to controller device driver instance.
+ * param[in] payload  Pointer to CCC payload.
+ *
+ * return:  The same as i3c_do_ccc()
+ *          0 If successful.
+ *          -EBUSY Bus is busy.
+ *          -EIO General Input / output error.
+ *          -EINVAL Invalid valid set in the payload structure.
+ *          -ENOSYS Not implemented.
+ */
+static int npcx_i3c_do_ccc(const struct device *dev, struct i3c_ccc_payload *payload)
+{
+	const struct npcx_i3c_config *config = dev->config;
+	int ret = 0;
+	struct i3c_reg *inst = config->base;
+	uint32_t intmask;
+	int xfered_len;
+
+	if (payload == NULL) {
+		return -EINVAL;
+	}
+
+	if (config->common.dev_list.num_i3c == 0) {
+		/*
+		 * No i3c devices in dev tree. Just return so
+		 * we don't get errors doing cmds when there
+		 * are no devices listening/responding.
+		 */
+		return 0;
+	}
```

Comment:
I'm not sure what the point of this is? I think someone want to see the nack on the i3c broadcast header if there are not i3c devices currently known in it's 'list.

I think you're doing this because of the `i3c_bus_init` in the driver init function so you will a return of `0` rather than an error. Would it make more sense to have it "not call" the `i3c_bus_init` if there is nothing in it's i3c device list instead?

Reply:
Done.
I have moved the target device checking before `i3c_bus_init` in `npcx_i3c_init`.